test/tstQueryTreeManufacturedSolution.cpp

```diff
@@ -88,11 +92,12 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(structured_grid, TreeTypeTraits,
   Kokkos::deep_copy(offset_host, offset);
 
   // we expect the collision list to be diag(0, 1, ..., nx*ny*nz-1)
-  for (int i = 0; i < n; ++i)
-  {
-    BOOST_TEST(indices_host(i) == i);
-    BOOST_TEST(offset_host(i) == i);
-  }
+  std::vector<int> indices_ref(n);
+  std::vector<int> offset_ref(n + 1);
+  std::iota(indices_ref.begin(), indices_ref.end(), 0);
+  std::iota(offset_ref.begin(), offset_ref.end(), 0);
+  BOOST_TEST(indices_host == indices_ref, tt::per_element());
+  BOOST_TEST(offset_host == offset_ref, tt::per_element());
```

Comment:
We should be comparing them as a "compressed storage"

Reply:
Updated.